usecases/user_usecase.go

```diff
@@ -26,6 +26,16 @@ func (usecase *UserUseCase) AddUser(ctx context.Context, createUser models.Creat
 		return models.User{}, errors.Wrap(models.BadParameterError, "Invalid role received")
 	}
 
+	if createUser.Role == models.TRANSFER_CHECK_USER && createUser.PartnerId == nil {
+		return models.User{}, errors.Wrap(models.BadParameterError,
+			"PartnerId is required for TRANSFER_CHECK_USER role")
+	}
+
+	if createUser.PartnerId != nil && createUser.Role != models.TRANSFER_CHECK_USER {
+		return models.User{}, errors.Wrap(models.BadParameterError,
+			"Partner Id is only allowed for TRANSFER_CHECK_USER role")
+	}
```

Comment:
I think splitting user roles per app and check on an array here, will be better.
It will allow to create differnet role types for TC (if the product ask for it) easilly

Reply:
what do you mean exactly, have a different usecase for the two apps ?

Reply:
More or less something like that, so if we create role management in transfercheck, it will not be a burdon (ATM, there is only user and api roles)
```suggestion
	if slices.Contains(models.GetValidTransfercheckUserRoles(), createUser.Role) && createUser.PartnerId == nil {
		return models.User{}, errors.Wrap(models.BadParameterError,
			"PartnerId is required for TRANSFER_CHECK_USER role")
	}

	if slices.Contains(models.GetValidMarbleUserRoles(), createUser.Role) && createUser.PartnerId != nil {
		return models.User{}, errors.Wrap(models.BadParameterError,
			"Partner Id is only allowed for TRANSFER_CHECK_USER role")
	}
```